File: Baekjoon/Greedy/14469.py
# https://www.acmicpc.net/problem/14469

import sys
input = sys.stdin.readline

n = int(input())
cow = [list(map(int, input().split())) for _ in range(n)]
cow.sort(key=lambda x: x[0])

time = 0

for i in range(n):
    if time < cow[i][0]:
        time = cow[i][0] + cow[i][1]
    else:
        time += cow[i][1]

print(time)

File: Baekjoon/Greedy/14469.cpp
// https://www.acmicpc.net/problem/14469

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n;
    vector<pair<int, int>> a;

    cin >> n;
    a.resize(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i].first >> a[i].second;
    }

    sort(a.begin(), a.end());
    int time = 0;
    for (int i = 0; i < n; i++) {
        if (time < a[i].first) {
            time = a[i].first;
        }
        time += a[i].second;
    }
    cout << time << '\n';

    return 0;
}